app/controllers/compute_resources_controller.rb

```diff
@@ -91,16 +91,20 @@ def destroy
   end
 
   def refresh_cache
-    if @compute_resource.respond_to?(:refresh_cache) && @compute_resource.refresh_cache
-      process_success(
-        :success_msg => _('Successfully refreshed the cache.'),
-        :success_redirect => @compute_resource
-      )
+    if @compute_resource.respond_to?(:refresh_cache)
+      if @compute_resource.refresh_cache
+        process_success(
+          :success_msg => _('Successfully refreshed the cache.'),
+          :success_redirect => @compute_resource
+        )
+      else
+        process_error(
+          :error_msg => _('Failed to refresh the cache.'),
+          :redirect => @compute_resource
+        )
+      end
     else
-      process_error(
-        :error_msg => _('Failed to refresh the cache.'),
-        :redirect => @compute_resource
-      )
+      process_error(:error_msg => 'Refreshing Cache is not supported for this compute resource.')
```

Comment:
Perhaps this should also be translated.
```suggestion
      process_error(:error_msg => N_("Cache refreshing is not supported for %s"), @compute_resource.provider_friendly_name)
```

Probably good to use the exact same translation string in the API to make life of translators easier :)